lib/IDE/CodeCompletionResultType.cpp

```diff
@@ -281,6 +308,11 @@ calculateMaxTypeRelation(Type Ty, const ExpectedTypeContext &typeContext,
                          const DeclContext &DC) {
   if (Ty->isVoid() && typeContext.requiresNonVoid())
     return TypeRelation::Invalid;
+  if (typeContext.getExpectedAttributeKinds()) {
+    return (getAttributeKinds(Ty) & typeContext.getExpectedAttributeKinds())
+               ? TypeRelation::Convertible
+               : TypeRelation::Invalid;
+  }
```

Comment:
I think I'd prefer to use 'Flair' for this purpose. Can we introduce a flair like `commonTypeAtCurrentPosition`, and use it with existing `rareTypeAtCurrentPosition`?

Reply:
I think that using type relation actually makes sense here because `commonTypeAtCurrentPosition` and `rareTypeAtCurrentPosition` would just be replicating functionality that we already have implemented for type relations. And a 'Convertible' type relation after a `@` means that the type is convertible (i.e. identical) to a type that e.g. has a `@propertyWrapper` attribute.

This would be a little bit more work implementation-wise because `calculateTypeRelation` currently only returns a `TypeRelation` and no flair.